### DISCUSSION TITLE
Sticky boundary for particles

### DISCUSSION THREAD

**Discussion by Misery7100** [epochpic/epoch#470] [Q&A]
At the moment I'm experimenting with magnetic field generation through interaction of fs laser pulse with nano-target (cluster, rod, hollow rod etc). In this situation, an important role is played by the electrical neutrality of the system in the process of interaction with the pulse and after, since a large number of electrons fly off. As result with open boundaries a "fake" charge might expected. Is there any solution like stick boundary for particles or workaround with injectors?

Many thanks in advance,
Lev

**Comment by Status-Mirror** [comment]
Hey Lev,

The EPOCH field-solver updates the electric and magnetic fields of the system using Maxwell's equations, by calculating local field gradients and current densities. The equation:

$\nabla \cdot \pmb{E} = \frac{\rho }{ \epsilon_0}$

is not used. I would guess absolute charges wouldn't affect things.

To test this - I recommend you run your simulation with two different vacuum sizes behind the target. The smaller-vacuum simulation will lose many ejected electrons quickly, while high energy electrons will persist in the larger-vacuum simulation longer. I expect the fields around the target to be the same in both simulations, provided both vacuum layers are sufficiently large. This would demonstrate the time at which electrons escape have no effect on the nano-target sheath fields.

Let me know if I'm wrong!

Hope this helps,
Stuart

**Reply by Misery7100** [reply]
That's right, no problem with the charging box. Thanks very much!

**Comment by Misery7100** [comment]
@Status-Mirror 

Hello Stuart

I checked outputs again and seems like particle outflow has a major impact on longitudinal magnetic field generated through interaction with circular polarized laser pulse (we expect magnetic dipole and theory shows it, but in computation result we see a significant positive skew due electrons outflow). Also I found some strange bug that appears only in case of simple outflow or open boundaries for fields related to fake sperical longitudinal components in corners of modeling box. If necessary I can share details to reproduce that.

Anyway as I know there is a version where "sticky" boundary condition for particles still exists (particles outgoing of box replaced with motionless particles at the box boundaries). Could you help to find this version?

Many thanks in advance,
Lev

**Reply by Misery7100** [reply]
Or if there is workaround to reproduce "sticky" boundary conditions in more recent versions - I'd be very appreciated for a guidance :)

**Reply by Status-Mirror** [reply]
Hey Lev, 

There's something like this in the code. The EPOCH `thermal` boundary (see [here](https://epochpic.github.io/documentation/input_deck/input_deck_boundaries.html#thermal_boundaries)) is intended for when a plasma touches the simulation boundaries. Particles are reflected from the boundary, but the energy of the reflected particle is sampled from a thermal distribution. The physical idea is that we have let the original particle escape, and we have replaced it with a backwards propagating particle from beyond the boundary. Numerically we conserve the macro-particle number, without reflecting any high-energy particles which would otherwise escape the simulation.

The energies of the reflected particles are sampled from a Maxwellian, using the initial plasma temperature for the boundary cell. It's possible that if the cell was a vacuum, the temperature would be zero, and the particle energy would be reset to zero when it touches the boundary?

Because it isn't intended for initially-vacuum boundaries, I can't predict how it will behave. Even if it does work, I'm not sure it will solve your weird magnetic field problem. I'm not yet convinced if charge depletion on a distant boundary would affect a local magnetic field component - I think the dimensionality of the code would affect that more (some field components in 2D may be weird c.f. 3D). It's worth a try though!

Hope this helps,
Stuart

P.S., I wasn't at Warwick in the early days of EPOCH, so I don't know of any versions which accepted "sticky" as a boundary condition.

**Reply by Misery7100** [reply]
The fact is that the problem is in 3D code 😄 But anyway thanks for your help. Regarding CPML - do you know working version? As in the release version it still works strange (quadratic or even exp field amplitude growth without a reason), the same for v4.17

**Reply by Status-Mirror** [reply]
Ahh, well there goes my hunch. I'd be surprised if your problem is due to charge depletion, but I'm becoming less confident... 

Unfortunately, CPML is not a part of the code I've worked with. Any CPML experts feel free to jump in here.

**Reply by Status-Mirror** [reply]
Another thought - from the perspective of EPOCH's field solver, there should be no difference between a particle stuck on the boundary, and one which has been removed from the simulation.

The only time particle information is used to update fields is in the current density. An immobile particle on the boundary and a removed particle both have zero current, so no impact on the fields. I don't expect this workaround-sticky-boundary model to fix your issue.

If you're finding these issues go away when you include a larger simulation domain, then maybe you just need a bigger domain to fully resolve the magnetic field structure? If the problems don't go away with a larger field, there may be another issue at play here -maybe in the simulation setup.
